src/handle.ml

```diff
@@ -125,15 +125,16 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
         let (s,_,_) as r = scope_rule ss r in
         if !(s.sym_def) <> None then
           fatal_no_pos "Symbol [%s] cannot be (re)defined." s.sym_name;
-        Sr.check_rule ss.builtins r; r
+        r
       in
       let rs = List.map handle_rule rs in
       (* Adding the rules all at once. *)
       let add_rule (s,h,r) =
         out 3 "(rule) %a\n" Print.pp_rule (s,h,r.elt);
         Sign.add_rule ss.signature s r.elt
       in
-      List.iter add_rule rs; (ss, None)
+      List.iter add_rule rs;
+      List.iter (Sr.check_rule ss.builtins) rs; (ss, None)
```

Comment:
SR must be checked BEFORE adding the rules. So these modifications should be reverted.

Reply:
When we check SR for rule fl -> r, we should consider all the rules defining f. That is why the SR should be checked in this way.

Reply:
I don't understand. Checking SR for a rule or set of rules should not depend on the rules themselves.